src/osd/SnapMapper.cc

```diff
@@ -546,10 +546,9 @@ void SnapMapper::get_objects_by_prefixes(
   vector<hobject_t> *out)
 {
   int r = 0;
+
   /// maintain the prefix_itr between calls to avoid searching depleted prefixes
-  for ( ;
-	prefix_itr != prefixes.end() && out->size() < max && r == 0;
-	++prefix_itr) {
+  for ( ; prefix_itr != prefixes.end();	prefix_itr++) {
```

Comment:
We must change the loop as it increments the prefix when reaching MAX.
This was not a problem for  the original code since the iterator was reset on each call, but with my code it means we will jump to the next iterator after every 2 objects

Reply:
I'm ok with changing the loop, I meant that a src code change shouldn't be part of an "adding unit tests" commit.